<?xml version="1.0" encoding="utf-8"?>

<!-- Tier1 Launch File -->

<launch>

    <!-- Launch Arguments -->    
    <arg name="record_bag"      default="false" />
    <arg name="save_folder"     default="$(find gesture_recorder)/workspace" />
    <arg name="save_images"     default="false" />


    <!-- Set ROS Parameters -->
    <param name="save_folder"   type="str"  value="$(arg save_folder)" />
    <param name="save_images"   type="bool"  value="$(arg save_images)" />


    <!--  SubSystem Setup: Vision Support -->
    <include file="$(find gesture_recorder)/launch/vision_system.launch" >
        <arg name="max_distance" default="3" /><!-- Max Depth Threshold (meters) -->
    </include>


    <!-- Open RQT for analysis with a custom perspective -->
   <node name = "RQT_TestViewer" pkg = "rqt_gui" type = "rqt_gui" respawn = "false" output = "screen" 
        args = "--perspective-file $(find gesture_recorder)/config/viewer.perspective"/>  


    <!-- Record ROS Bag for Debugging Help -->
    <node pkg="rosbag" type="record" name="rosbag_record_cam"
        args="record -a -o '$(find gesture_recorder)/workspace/'"
        if="$(arg record_bag)" />


    <!-- Launch TopLevel Python Script -->
    <node pkg="gesture_recorder" type="recorder.py" name="recorder_obj" output="screen" launch-prefix="gnome-terminal --command" />

</launch>